api/src/main/java/org/pmiops/workbench/workspaces/WorkspaceService.java

```diff
@@ -67,4 +70,20 @@ WorkspaceACLUpdate updateFirecloudAclsOnUser(
       WorkspaceAccessLevel updatedAccess, WorkspaceACLUpdate currentUpdate);
 
   Workspace setPublished(Workspace workspace, String publishedWorkspaceGroup, boolean publish);
+
+  List<UserRecentWorkspace> getRecentWorkspaces();
+
+  UserRecentWorkspace updateRecentWorkspaces(
+      long workspaceId, long userId, Timestamp lastAccessDate);
+
+  UserRecentWorkspace updateRecentWorkspaces(long workspaceId);
+
+  boolean maybeDeleteRecentWorkspace(long workspaceId);
+
+  List<RecentWorkspace> buildRecentWorkspaceList(List<UserRecentWorkspace> userRecentWorkspaces);
```

Comment:
My understanding was that we wanted Controllers to be very thin wrappers around service calls. Is your reasoning for saying these belong in the controller because they construct the actual objects that get returned to the client? I can certainly move this up in any case

Reply:
Sorry I missed this one. I don't really share that world view - but I think the comment stands independently.

If a method would only conceivably be used in one place, it makes sense to localize it there, I think that's the case for the controller which calls this method. Alternatively, you could just update the method signature to return the API model instead of the DB model (and make these methods private to the service) - right now the interface has a mix of DB and API.